### PR TITLE
Add React authorization button to navbar

### PR DESCRIPTION
**Pull request by Richie78321**
* Add a React-based authorization button that fetches authorization data from the authorization servlet and updates its appearance to either be a "Log In" button or a "Log Out" button.
* Add this authorization button to the navbar for easy website-wide access.
* Update the main style sheet `style.css` to be more permitting of other buttons other than those on the main home page.

The intended flow is for a new user to sign up with this button, from which they are redirected (by the login page) to their profile, where they can set data about themselves and link their GitHub accounts.

### CODE REVIEW

**Comment by Richie78321** [comment]
Waiting for #50 to be merged into master to clean up the commits here.

File: src/main/webapp/js/common/react/components/AuthButton.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    this.state = {
+      isFetching: true,
+    };
+  }
+
+  /**
+   * Loads user authorization data after React has initialized the component.
+   * @override
+   */
+  componentDidMount() {
```
Comment (Sami-2000) [inline src/main/webapp/js/common/react/components/AuthButton.js:27]: Q: Does this function only call another function? (if so, could you just move the body of 'fetchAuthorizationStatus' here? or will you need that function other places)

File: src/main/webapp/style.css [hunk 2]
```diff
@@ -11,10 +11,25 @@ a:hover,
...
   color: white;
-  width: 70%;
+}
+
+.btn.disabled {
+  background-color: rebeccapurple;
+  color: white;
+}
+
+.btn-shadow:hover {
```
Comment (Sami-2000) [inline src/main/webapp/style.css:31]: Q: To clarify: now Obi and I need to add this class if we want box shadows on hover? (if we need to make this change plz throw it in 3 peas chat when u merge so he and I can change our code accordingly)

File: src/main/webapp/js/common/react/components/AuthButton.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    });
+  }
+
+  /**
+   * Renders the Login/Logout button. If the authorization data is still
+   * loading, the button is disabled.
+   * @return {React.Component}
+   * @override
+   */
+  render() {
```
Comment (Sami-2000) [inline src/main/webapp/js/common/react/components/AuthButton.js:58]: Trailing spaces seem like the nittiest nit, but I suppose if you push this branch again you may as well remove them so we can get rid of these linting messages.

Comment (billyeh) [reply src/main/webapp/js/common/react/components/AuthButton.js:58]: Yes please. Let's get rid of all linter errors before merging anything to master :)

Comment (Richie78321) [reply src/main/webapp/js/common/react/components/AuthButton.js:58]: Sgtm! Thanks for the reminder. Will update those now.

File: src/main/webapp/js/common/react/components/AuthButton.js (hunk 1 above)
Comment (Richie78321) [reply src/main/webapp/js/common/react/components/AuthButton.js:27]: Good point. I think this is a remnant of thinking I was going to need it more than once. Will update!

File: src/main/webapp/style.css (hunk 2 above)
Comment (Richie78321) [reply src/main/webapp/style.css:31]: Yeah you will. Sorry for doing this but it'll be nice to have the option of a shadow on some buttons. I'll send out a chat now.

**Pull request merged**